hpedockerplugin/volume_manager.py

```diff
@@ -1357,36 +1395,91 @@ def mount_volume(self, volname, vol_mount, mount_id):
+                    old_path_info = vol.get('old_path_info', [])
+
+                    # Check if old_node_id is already present in old_path_info
+                    # If found, replace it by removing the existing ones and
+                    # appending the new one
+                    if old_path_info:
+                        LOG.info("Old path info found! Removing any "
+                                 "duplicate entries...")
+                        # This is a temporary logic without a break statement
+                        # This is required to remove multiple duplicate tuples
+                        # (node_id, path_info) i.e. entries with same node_id
+                        # Later on
+                        updated_list = []
+                        for opi in old_path_info:
+                            node_id = opi[0]
+                            if old_node_id == node_id:
+                                LOG.info("Found old-path-info tuple "
+                                         "having node-id %s for volume %s. "
+                                         "Skipping it..."
+                                         % (node_id, volname))
+                                continue
+                            updated_list.append(opi)
+                        old_path_info = updated_list
+
+                    old_path_info.append((old_node_id, path_info))
+                    self._etcd.update_vol(volid, 'old_path_info',
+                                          old_path_info)
 
                 node_mount_info = {self._node_id: [mount_id]}
                 LOG.info("New node_mount_info set: %s" % node_mount_info)
+            elif flag == VolumeOwnedAndNotMounted:
+                LOG.info("This might be the case of reboot...")
+                LOG.info("Volume %s is owned by this node %s but it is not "
+                         "in mounted state" % (volname, self._node_id))
+                # We need to simply mount the volume using the information
+                # in ETCD
+                path_info = self._etcd.get_path_info_from_vol(vol)
+                if path_info:
+                    path = path_info['path']
+                    mount_dir = path_info['mount_dir']
+                    if 'dm-' in path and \
+                            fileutil.check_if_file_exists(mount_dir) and \
+                            fileutil.check_if_file_exists(path):
+                        LOG.info("Case of reboot confirmed! Mounting device "
+                                 "%s on path %s" % (path, mount_dir))
+                        try:
+                            fileutil.mount_dir(path, mount_dir)
+                        except Exception as ex:
+                            msg = "Mount volume failed: %s" % \
+                                  six.text_type(ex)
+                            LOG.error(msg)
+                            self._rollback(undo_steps)
```

Comment:
Will this give any error , here undo_steps is an empty list. Shall we remove this _rollback(undo_steps)
if it doesnot raise any exception i am okay with this change.

Rest of changes are okay for me.

Reply:
It won't cause any issues. I would suggest to retain it just in case some other action is required above in the future needing rollback will have it without any change.